File: AID/python/pyAID/tests/test_splits.py
import numpy as np

from pyAID.utils import find_best_split, sse_from_stats


def test_best_split_on_simple_step():
    X = np.array([[0.0], [1.0], [2.0], [3.0]])
    y = np.array([0.0, 0.0, 1.0, 1.0])
    parent_sse = sse_from_stats(np.sum(y), np.sum(y * y), len(y))

    candidate = find_best_split(X, y, parent_sse=parent_sse, min_child_size=1)
    assert candidate is not None
    assert candidate.feature_index == 0
    assert 1.0 < candidate.threshold < 2.0
    assert candidate.gain > 0

File: AID/r/RAID/tests/testthat/test_splits.R
library(testthat)

test_that("best split found on step function", {
  X <- matrix(c(0, 1, 2, 3), ncol = 1)
  y <- c(0, 0, 1, 1)
  parent_sse <- RAID:::sse_from_stats(sum(y), sum(y * y), length(y))
  cand <- RAID:::find_best_split(X, y, parent_sse = parent_sse, min_child = 1)
  expect_false(is.null(cand))
  expect_gt(cand$gain, 0)
  expect_true(cand$threshold > 1 && cand$threshold < 2)
})
